server/src/main/java/org/cloudfoundry/identity/uaa/user/VmidentityUserDatabase.java

```diff
@@ -45,51 +45,37 @@ public VmidentityUserDatabase(CasIdmClient casIdmClient, Set<String> defaultAuth
         // mergeAuthorities(this._adminAuthorities, UaaAuthority.ADMIN_AUTHORITIES);
         // mergeAuthorities(this._regularAuthorities, UaaAuthority.USER_AUTHORITIES);
     }
-    // IdentityZoneHolder.get() gives current tenant
-    // IdentityZone.getUaa() - gets system tenant name
 
     @Override
     public UaaUser retrieveUserByName(String username, String origin) throws UsernameNotFoundException {
         try {
             String tenant = VmidentityUtils.getTenantName(this._idmClient.getSystemTenant());
```

Comment:
Just a note - this winds up creating an extra call to IDM every time to fetch the system tenant's name, even in scenarios where we don't need it (e.g. not using the default uaa-zone). I've got a change in the bit I'm working on right now where you pass the client instead of the system tenant name, so it only calls out when it's necessary.